### DISCUSSION TITLE
Next.js v13.4.9 - reading the body of a POST request in API routes

### DISCUSSION THREAD

**Discussion by jdamon96** [vercel/next.js#52694] [Help]
### Summary

I'm currently running into the following JSON error on Next.js v13.4.9 whenever I try to unpack the request body in my POST req API route handler: 

`SyntaxError: Unexpected end of JSON input`

### Additional information

I call my API route in a form component with the following code:

```js
const onSubmit = async () => {
    const data = {
      field: <value>,
    };

    const api_url = process.env.URL + "/app/api";

    const api_req_options = {
      method: "POST",
      headers: {
        "content-type": "application/json",
      },
      body: JSON.stringify(data),
    };

    const res = await fetch(api_url, api_req_options);
}
```

Where this is my route handler code:

```js
// /app/api/route.ts
import { NextRequest, NextResponse } from "next/server";


export async function POST(req: NextRequest, res: NextResponse) {
  try {
    const body = await req.json();
    console.log(body);  
    return NextResponse.json({ message: "Hello, world!" });
  } catch (err) {
    console.error(err);
    return NextResponse.error();
  }
}
```

And every time, the app crashes because the route handler fails with the following error:

```
- error SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at parseJSONFromBytes (node:internal/deps/undici/undici:6571:19)
    at successSteps (node:internal/deps/undici/undici:6545:27)
    at node:internal/deps/undici/undici:1211:60
    at node:internal/process/task_queues:140:7
    at AsyncResource.runInAsyncScope (node:async_hooks:203:9)
    at AsyncResource.runMicrotask (node:internal/process/task_queues:137:8)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

I'm usually inclined to think this is a silly user-error on my part, but I've seen similar issues in both this thread here, as well as [this issue that documents the same "problem" but for DELETE routes](https://github.com/vercel/next.js/issues/48096) (I put "problem" in quotations bc, as a poster in that thread notes, DELETE routes technically aren't supposed to support request bodies). 

A POST request should definitely support a body field in the request, though, and I can't figure out if I'm doing something wrong here or this is a Next issue
```

**Comment by MichaelJuannn** [comment]
use req.formData() to get data from form data
